### ISSUE TITLE
Demo creation fails if site name does not start with a letter character (incl CWP)

### ISSUE THREAD

**Issue by brynwhyman** [silverstripe/bambusa-installer#111]
Example error message: `(value is \"6xpertscom-sscom\"): only lowercase letters, numbers, underscores and dashes are allowed, must start with a letter"`

This is a limitation of the underlying demo creation infrastructure. We see the same problem with the CWP demo: https://github.com/silverstripeltd/cwp-demo/issues/18

It seems that a quick win here would be some validation on the demo request from. 

There's a related issue that could also benefit from front-end validation: https://github.com/silverstripe/bambusa-installer/issues/87

## AC's
- Where a user attempts to create a demo where the name begins with a non-alpha character, the demo is still successfully created, or the user is given information to change the name.
- This issue exists for the CWP demo as well. The issue should be resolved for CWP as part of this issue too
- Ensure that the team (mostly) responsible for s.com is consulted as part of the deployment required for the Bambusa fix

## PR's

- [x] https://github.com/silverstripeltd/silverstripe-demostacker/pull/53 - add better validation
- [x] https://github.com/silverstripeltd/silverstripe.com/pull/186 - use `domain_prefix`
- [x] https://github.com/silverstripeltd/cwp-demo/pull/31 - use `domain_prefix`
- [x] https://www.silverstripe.com/silverstripe-cms/demo/ deployed ([sscom2](https://silverstripe.cloud/naut/project/sscom2/environment/prod/overview/))
    - [x] https://github.com/silverstripeltd/silverstripe.com/pull/187/files - fix circleCI on silverstripe.com 
    - [x] https://github.com/silverstripeltd/silverstripe.com/pull/188 - update composer demostacker
- [x] https://demo.cwp.govt.nz/ deployed ([cwpdemos](https://dash.cwp.govt.nz/naut/project/cwpdemos/environment/prod/overview) / [ssdemo](https://silverstripe.cloud/naut/project/ssdemo/environment/prod/overview/))
    - [x] https://github.com/silverstripeltd/cwp-demo/pull/32 - update composer demostacker

**Comment by maxime-rainville** [comment]
We're adding a suffix to the demo name based on what environment the demo is created on. This is meant to avoid collisions between CWP and DotCOM demoes.

A simple fix could be to make it a prefix instead of a suffix.

**Comment by brynwhyman** [comment]
We've got [the same problem with the CWP demo](https://github.com/silverstripeltd/cwp-demo/issues/18) so it seems we should do the same fix for both. I don't believe the CWP demos have a prefix. 

If we fix this with form validation, we could probably squeeze the following in as well: https://github.com/silverstripe/bambusa-installer/issues/87

**Comment by bergice** [comment]
Added new PR's to update sites with merged demostacker changes and to fix circleCI builds on silverstripe.com.

**Comment by brynwhyman** [comment]
Blocked by: https://silverstripe.freshdesk.com/a/tickets/67189

The latest suggestion is to do a re-deploy of the base stack holding the ss.com vstack. This could result in a PHP upgrade for other projects, so we'll need to make sure that has no adverse affects. I can have a look at this and track someone down to help approve that.

**Comment by brynwhyman** [comment]
We've received approval from Olivia to proceed with a base stack deployment: https://silverstripeltd.slack.com/archives/C0XL54DEH/p1598910385007000?thread_ts=1598910107.006700&cid=C0XL54DEH
